### PR TITLE
mgmt/mcumgr: Move OS specific source code for OS and FS groups

### PR DESCRIPTION
**Pull request by de-nordic**
Two commits that remove zephyr_ prefixes, move OS (Zephyr) specific functions to main source files and add "static" specifier to them,  for OS and FS group, as there is no longer need to keep them in separate files.

### CODE REVIEW

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c [hunk 1]
```diff
@@ -72,6 +73,114 @@ static const struct mgmt_handler fs_mgmt_handlers[];
...
+	if (rc < 0) {
+		return MGMT_ERR_EUNKNOWN;
+	}
+
+	return 0;
+}
+
+static int
+fs_mgmt_write(const char *path, size_t offset, const void *data,
+				   size_t len)
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c:139]: Alignment (can probably go up a line)

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c [hunk 2]
```diff
@@ -72,6 +73,114 @@ static const struct mgmt_handler fs_mgmt_handlers[];
...
+	if (dirent.type != FS_DIR_ENTRY_FILE) {
+		return MGMT_ERR_EUNKNOWN;
+	}
+
+	*out_len = dirent.size;
+
+	return 0;
+}
+
+static int
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c:101]: Does it make sense to keep these on separate lines anymore?

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c [hunk 3]
```diff
@@ -72,11 +73,115 @@ static const struct mgmt_handler fs_mgmt_handlers[];
...
+
+	if (rc < 0) {
+		return MGMT_ERR_EUNKNOWN;
+	}
+
+	return 0;
+}
+
+static int fs_mgmt_write(const char *path, size_t offset,
+			 const void *data, size_t len)
```
[the remark this answers is not in the document]
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c]: The functions are already long and indented enough.

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c [hunk 4]
```diff
@@ -266,8 +267,22 @@ static int os_mgmt_taskstat_read(struct mgmt_ctxt *ctxt)
...
  * Command handler: os reset
  */
-static int
-os_mgmt_reset(struct mgmt_ctxt *ctxt)
+static void os_mgmt_reset_work_handler(struct k_work *work)
+{
+	sys_reboot(SYS_REBOOT_WARM);
+}
+
+K_WORK_DEFINE(os_mgmt_reset_work, os_mgmt_reset_work_handler);
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:275]: Put at top of the file with variable declaration (use prototype for function)

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c [hunk 5]
```diff
@@ -266,8 +267,22 @@ static int os_mgmt_taskstat_read(struct mgmt_ctxt *ctxt)
...
+
+static void os_mgmt_reset_cb(struct k_timer *timer)
+{
+	/* Reboot the system from the system workqueue thread. */
+	k_work_submit(&os_mgmt_reset_work);
+}
+
+static K_TIMER_DEFINE(os_mgmt_reset_timer, os_mgmt_reset_cb, NULL);
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:283]: As above

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c [hunk 6]
```diff
@@ -266,8 +267,22 @@ static int os_mgmt_taskstat_read(struct mgmt_ctxt *ctxt)
 /**
  * Command handler: os reset
  */
-static int
-os_mgmt_reset(struct mgmt_ctxt *ctxt)
+static void os_mgmt_reset_work_handler(struct k_work *work)
+{
+	sys_reboot(SYS_REBOOT_WARM);
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:272]: This loses the ifdef protection from the cmake file

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c [hunk 7]
```diff
@@ -159,11 +215,73 @@ fs_mgmt_file_download(struct mgmt_ctxt *ctxt)
 	return ok ? MGMT_ERR_EOK : MGMT_ERR_EMSGSIZE;
 }
 
+static int fs_mgmt_write(const char *path, size_t offset,
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c:218]: Nothing wrong with these functions, but, the zephyr_fs_mgmt.c file does not appear to have been removed in this commit when it should be

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c (hunk 6 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c:280]: It is under the #ifdef CONFIG_REBOOT, the same used to include the zephyr_os_mgmt.c.

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c (hunk 5 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c]: Done

File: subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c (hunk 4 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c]: Done

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**